integration/ledger/snapshot_test.go

```diff
@@ -0,0 +1,816 @@
+
+		verifyQSCC(setup.network, org4peer0, testchannelID, blockNum2, txidBeforeSnapshot)
+		verifyChannelConfigUpdate(setup.network, setup.orderer, org4peer0, testchannelID)
+
+		// test 8: verify chaincode upgrade and install after bootstrapping
+		By("upgrading chaincode to version 2.0 on all peers after bootstrapping from snapshot")
+		newlifecycleChaincode.Version = "2.0"
+		newlifecycleChaincode.Sequence = "2"
+		nwo.DeployChaincode(setup.network, testchannelID, setup.orderer, newlifecycleChaincode)
+
+		By("deploying a new chaincode on all the peers after bootstrapping from snapshot")
+		cc2 := nwo.Chaincode{
+			Name:            "kvexecutor",
+			Version:         "1.0",
+			Path:            components.Build("github.com/hyperledger/fabric/integration/chaincode/kvexecutor/cmd"),
+			Lang:            "binary",
+			SignaturePolicy: `OR ('Org1MSP.member','Org2MSP.member', 'Org3MSP.member', 'Org4MSP.member')`,
+			PackageFile:     filepath.Join(setup.testDir, "kvexcutor20.tar.gz"),
+			Label:           "kvexecutor-20",
+			Sequence:        "1",
+		}
+		nwo.DeployChaincode(setup.network, testchannelID, setup.orderer, cc2)
+
+		By("invoking the new chaincode")
+		kvdata := []kvexecutor.KVData{
+			{Key: "key1", Value: "value1"},
+			{Key: "key2", Value: "value2"},
+		}
+		invokeAndQueryKVExecutorChaincode(setup.network, setup.orderer, testchannelID, cc2, kvdata, setup.network.PeersWithChannel(testchannelID)...)
+	})
+})
+
+func configPeerWithCouchDB(s *setup, peer *nwo.Peer) ifrit.Process {
+	couchDB := &runner.CouchDB{}
+	couchProc := ifrit.Invoke(couchDB)
+	Eventually(couchProc.Ready(), runner.DefaultStartTimeout).Should(BeClosed())
+	Consistently(couchProc.Wait()).ShouldNot(Receive())
+
+	core := s.network.ReadPeerConfig(peer)
+	core.Ledger.State.StateDatabase = "CouchDB"
+	core.Ledger.State.CouchDBConfig.CouchDBAddress = couchDB.Address()
+
+	By("configuring peer to couchdb address " + couchDB.Address())
+	s.network.WritePeerConfig(peer, core)
+
+	return couchProc
+}
+
+// initAndStartMultiChannelNetwork creates a network with 2 orgs (Org1, Org2), 2 peers (peer0, peer1) per org,
+// and 3 channels (testchannel, testchannel2, testchannel3).
+// peer0 in both org: join testchannel and testchannel2
+// peer1 in org1: join no channel
+// peer1 in org2: join testchannel2
+func initAndStartMultiChannelNetwork() *setup {
+	var err error
+	testDir, err := workDir("/tmp", "snapshot")
+	Expect(err).NotTo(HaveOccurred())
+
+	client, err := docker.NewClientFromEnv()
+	Expect(err).NotTo(HaveOccurred())
+
+	config := nwo.BasicSolo()
+
+	config.Channels = []*nwo.Channel{
+		{Name: testchannelID, Profile: "TwoOrgsChannel"},
+	}
+
+	for _, peer := range config.Peers {
+		peer.Channels = []*nwo.PeerChannel{
+			{Name: testchannelID, Anchor: true},
+		}
+	}
+
+	for _, org := range config.Organizations {
+		if org.Name == "Org1" || org.Name == "Org2" {
+			org.Users = 3
+		}
+	}
```

Comment:
Is this required now?

Reply:
Not required now, remove it from this PR. Still will need more users for pvtdata tests.